lib/ansible/modules/database/postgresql/postgresql_pg_hba.py

```diff
@@ -0,0 +1,673 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+'''
+This module is used to manage postgres pg_hba files with Ansible.
+'''
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: postgresql_pg_hba
+short_description: Adds, removes or modifies an rule in a pg_hba file.
+description:
+   - The fundamental function of the module is to create, or delete lines in pg_hba files.
+   - The lines in the file should be in a typical pg_hba form and lines should be unique per key (type, databases, users, source).
+     If they are not unique and the SID is 'the one to change', only one for C(state=present) or none for C(state=absent) of the SID's will remain.
+extends_documentation_fragment: files
+version_added: "2.8"
+options:
+  address:
+    description:
+      - The source address/net where the connections could come from.
+      - Will not be used for entries of I(type)=C(local).
+      - You can also use keywords C(all), C(samehost), and C(samenet).
+    default: "samehost"
+    aliases: [ 'source', 'src' ]
```

Comment:
```suggestion
    type: str
    aliases: [ 'source', 'src' ]
```
It's a strong recommendation to use type for each option